client/a8c-for-agencies/sections/sites/needs-setup-sites/site-configurations-modal/index.tsx

```diff
@@ -49,50 +49,74 @@ export default function SiteConfigurationsModal( {
 		Object.keys( dataCenterOptions ) as Array< keyof typeof dataCenterOptions >
 	 ).map( ( key ) => (
 		<option key={ key } value={ key }>
-			{ dataCenterOptions[ key ] }
+			{ dataCenterOptions[ key as keyof typeof dataCenterOptions ] }
 		</option>
 	) );
 
+	const onSiteNameKeyDown = ( event: React.KeyboardEvent ) => {
+		if ( event.key === 'Enter' ) {
+			event.preventDefault();
+		}
+	};
+
+	const onSubmit = ( event: React.FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+		const formData = new FormData( event.currentTarget );
+		const phpVersion = formData.get( 'php_version' ) as string;
+		const primaryDataCenter = ( formData.get( 'primary_data_center' ) as string ) || undefined;
+		const params = {
+			site_name: siteName.siteName,
+			php_version: phpVersion,
+			primary_data_center: primaryDataCenter,
+			is_fully_managed_agency_site: allowClientsToUseSiteHelpCenter,
+		};
+		// eslint-disable-next-line no-console
+		console.log( params );
```

Comment:
This will help us on next steps. Leveraging the submit for event helps us to not assign a React state for every form field.